src/react/AlertManager.js

```diff
@@ -0,0 +1,270 @@
+    (message = "", options = {}) => {
+      options.variant = "warning";
+      options.timeout = 0;
+      return show(message, options);
+    },
+    [show]
+  );
+
+  const showError = useCallback(
+    (message = "", options = {}) => {
+      options.variant = "danger";
+      options.timeout = 0;
+      return show(message, options);
+    },
+    [show]
+  );
+
+  const showInfo = useCallback(
+    (message = "", options = {}) => {
+      options.variant = "info";
+      options.timeout = timeout || 0;
+      return show(message, options);
+    },
+    [show]
+  );
+
+  return {
+    alerts,
+    close,
+    closeAll,
+    showSuccess,
+    showWarning,
+    showError,
+    showInfo,
+  };
+};
+
+const AlertsBanner = ({ alerts: currentAlerts, setNotifier }) => {
+  const [alerts, setAlerts] = useState(currentAlerts);
+  useEffect(() => setAlerts(currentAlerts), [setAlerts, currentAlerts]);
+  setNotifier && setNotifier(setAlerts);
+  if (!alerts || alerts.length === 0) {
+    return null;
+  }
+  return (
+    <Row>
+      {alerts.map((alert) => (
+        <Col key={alert.id} xs={12}>
+          <Alert
+            variant={alert.options.variant}
+            onClose={alert.close}
+            dismissible
+          >
+            {alert.message}
+          </Alert>
+        </Col>
+      ))}
+    </Row>
+  );
+};
+
+const AlertsSnackbar = ({ alerts }) => {
+  if (!alerts || alerts.length === 0) {
+    return null;
+  }
+  return (
+    <Container fluid className="fixed-bottom p-3">
+      {alerts.map((alert) => (
+        <Toast key={alert.id} onClose={alert.close} className={`mx-auto`}>
+          <Toast.Header className={`bg-${alert.options.variant} text-light`}>
+            <span className="mx-auto">
+              {new Date(alert.timestamp).toLocaleString()}
+            </span>
+          </Toast.Header>
+          <Toast.Body>{alert.message}</Toast.Body>
+        </Toast>
+      ))}
+    </Container>
+  );
+};
+
+const GlobalAlertsContext = createContext();
+
+const GlobalAlertsProvider = ({ children, ...props }) => {
+  const root = useRef(null);
+  const alertContext = useRef(null);
+  const {
+    alerts,
+    close,
+    closeAll,
+    showSuccess,
+    showWarning,
+    showError,
+    showInfo,
+  } = useAlertsContext({ timeout: 5000 });
+
+  useEffect(() => {
+    root.current = document.createElement("div");
+    root.current.id = "__alert-manager__";
+    document.body.appendChild(root.current);
+    return () => {
+      if (root.current) document.body.removeChild(root.current);
+    };
+  }, []);
+
+  const Alerts = useMemo(() => {
+    return () => <AlertsSnackbar alerts={alerts} />;
+  }, [alerts]);
+
+  alertContext.current = {
+    close,
+    closeAll,
+    showSuccess,
+    showWarning,
+    showError,
+    showInfo,
+  };
+
+  return (
+    <GlobalAlertsContext.Provider value={alertContext} {...props}>
+      {children}
+      {root.current && createPortal(<Alerts />, root.current)}
```

Comment:
Here you have a context provider that also injects elements in the DOM. If on a later that we will move alerts on a different level of the layout, this code will need some heavy refactoring.

Reply:
Good point. I'll refactor it so we can place the Alerts where we need them.